src/dialects/abstract/query-interface.d.ts

```diff
@@ -444,8 +444,16 @@ export class QueryInterface {
   /**
    * Removes an index of a table
    */
-  removeIndex(tableName: TableName, indexName: string, options?: QueryInterfaceIndexOptions): Promise<void>;
-  removeIndex(tableName: TableName, attributes: string[], options?: QueryInterfaceIndexOptions): Promise<void>;
+  removeIndex(
+    tableName: TableName,
+    indexName: string,
+    options?: QueryInterfaceIndexOptions & RemoveIndexQueryOptions
```

Comment:
I think it complains here when you make `RemoveIndexQueryOptions` an interface right? You need to extend it, so make a new `QueryInterfaceRemoveIndexOptions` interface extending these two

Reply:
As above, creating a separate interface still produces the same error.

Reply:
Yes the problem really is that the method incorrectly expects a record, which have an index signature that interfaces don't have by default (and that we shouldn't add to our interfaces of course)

Reply:
Okay, I've added the interface and updated the type of options for rejectInvalidOptions to object.